pages/index.tsx

```diff
@@ -26,23 +27,42 @@ export default function Home({
   const [receiverError, setReceiverError] = useState(false);
   const [newReceiver, setNewReceiver] = useState('');
   const [newGiftName, setNewGiftName] = useState('');
-  const [isDeleteModalOpen, setIsDeleteModalOpen] = useState(false);
-  const [deleteModalGiftData, setDeleteModalGiftData] = useState<Gift>();
-  const [isEditModalOpen, setIsEditModalOpen] = useState(false);
-  const [editModalGiftData, setEditModalGiftData] = useState<Gift>();
+
   const [showUserWindow, setShowUserWindow] = useState(false);
 
+  const giftQuery = useQuery({
+    queryKey: ['loadingGifts'],
+    queryFn: async () => await fetchGifts(),
+    refetchOnWindowFocus: false,
+    refetchOnMount: false,
+    retry: false,
+  });
+
+  async function fetchGifts() {
+    try {
+      const gifts = await getAllGifts();
+      setGiftData(gifts);
+      return gifts;
+    } catch (e) {
+      handleErrorToast(handleError(e));
+      throw new Error('Palvelinvirhe! Lisätietoja kehittäjäkonsolissa');
+    }
+  }
+
   useEffect(() => {
     console.log('effect');
+    /*
     async function fetchGifts() {
       try {
         const gifts = await getAllGifts();
-        setGiftData(gifts);
+        console.log(gifts);
+        //setGiftData(gifts);
       } catch (e) {
         handleErrorToast(handleError(e));
       }
     }
     void fetchGifts();
+    */
```

Comment:
Tämä poistettu käytöstä tähän hätään, testauksia

Reply:
Tämän `useEffet`in voi nyt poistaa kokonaan kun react query otettu käyttöön.